pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -537,8 +551,8 @@ func (lbaas *LbaasV2) UpdateLoadBalancer(clusterName string, service *api.Servic
 
 	// Compose Set of member (addresses) that _should_ exist
 	addrs := map[string]empty{}
-	for _, nodeName := range nodeNames {
-		addr, err := getAddressByName(lbaas.compute, types.NodeName(nodeName))
+	for _, node := range nodes {
+		addr, err := nodeAddressForLB(node)
```

Comment:
I don't think we do... is GCE filtering them @bprashanth ?

In any case, that should be a separate issue / PR, not snuck into a refactor PR :-)